highfidelity/hifi
# Clara.io marketplace improvements

**NEXT STEP involves updating the UX**

Testing

 * Make sure Settings>developmer menus and Settings>Advanced menus are on.
 * Go to Developer>Debug defaultScripts.js and make sure it is checked
 * Edit >Running scripts. Reload default scripts. 
 * Close "marketplace.js" and add "mmarketplaces.js"

**OR**

Replace defaultScripts.js with this [defaultScripts.zip](https://github.com/highfidelity/hifi/files/643377/defaultScripts.zip)


Test:
- Add asset from Clara.io by going to Marketplace and navigating to the Clara marketplace
- Error dialog if try to download Clara asset and don't have "Write Assets" permission on the domain.
- Error dialog after download if don't have rez rights so model isn't added to world.
- Adding same asset from Clara.io creates a duplicate file and mapping (number appended to original name).
- Collision shape type of added model is hull per submesh.
- All Asset Browser functionality continues to work.

Check keyboard raising and lowering (in HMD view) in dialogs:
- Login dialog
- Address bar dialog
- Browser popped up from address bar's card details (click on "camera" or "person" icon then click on social media share button to open in new browser window)
- Running Scripts
- Entities editor
- Settings dialogs
- Avatar Browser from Avatar settings
- Attachments dialogs
- Attachments browser from Attachments dialog
- Browser
- Web entity
- Marketplace
- Add asset to world
- Rename asset
- Export entities

Check the dialogs in the following scripts still work OK (if these scripts are still used):
- playaPerformanceTest.js
- qmlTest.js
- textureStress.js
- toolWindowStressTest.js
- snapshot.js
- marketplace.js
